### ISSUE TITLE
Possibility of specifying LazyElement type for easy Tree based comparisons between two Json objects?

### ISSUE THREAD

**Issue by alphahelix00** [doubledutch/LazyJSON#12]
Would it be possible to specify whether the current Json element is a leaf, object, or array? I am trying to traverse the root Json object as a tree, comparing between Json outputs from a production vs. staging environment example. So I'd like to create a library using LazyJSON that compares for differences between leaves in the production vs staging trees.

I guess this would also require being able to return a LazyNode for keySets.

**Comment by kasperjj** [comment]
There is not currently any way to do this from LazyElement, but from both LazyObject and LazyArray you can ask for the type of each of its child nodes using getType. I will add a getType method to LazyElement soon that will let you deduce wether its an object or an array, but for now you can test this by using instanceof:
```java
LazyElement elm;
if(elm instanceof LazyObject){
    // This is an object
}else if(elm instanceof LazyArray){
    // This is an array
}else{
    // We shouldn't currently get here since LazyElement will always be either an array or an object.
}
````
LazyElements are never leaves, they are always objects or arrays. Once you have figured out what your root element is, you should be able to traverse the tree easily using getType. Let me know if you open source your tool, sounds interesting and useful!

**Comment by kasperjj** [comment]
Fixed in 1.2.7, should be available in maven within 24 hours.
